connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Loggers.java

```diff
@@ -198,26 +194,37 @@ org.apache.logging.log4j.Logger rootLogger() {
     }
 
     private void setLevel(org.apache.logging.log4j.Logger logger, Level level) {
-        String loggerName = logger.getName();
-        LoggerContext context = (LoggerContext) LogManager.getContext(false);
-        LoggerConfig loggerConfig = context.getConfiguration().getLoggerConfig(loggerName);
-        Level currentLevel = loggerConfig.getLevel();
+        Level currentLevel = logger.getLevel();
 
         if (level.equals(currentLevel)) {
-            log.debug("Skipping update for logger {} since its level is already {}", loggerName, level);
+            log.debug("Skipping update for logger {} since its level is already {}", logger.getName(), level);
             return;
         }
 
-        log.debug("Setting level of logger {} (excluding children) to {}", loggerName, level);
-        Configurator.setLevel(loggerName, level);
-        lastModifiedTimes.put(loggerName, time.milliseconds());
+        log.debug("Setting level of logger {} (excluding children) to {}", logger.getName(), level);
+        Configurator.setLevel(logger.getName(), level);
```

Comment:
according to https://github.com/apache/logging-log4j2/issues/2281#issue-2130099964, this can change all childs' level - that means the changed logger could have no last modified time

Maybe we need to trace all loggers the changes manually

Reply:
Interesting...
I will take a look